### PR TITLE
Add /supervisor/available_updates endpoint

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1135
- Link to cli pull request: https://github.com/home-assistant/cli/pull/319

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/supervisor.py [hunk 1]
```diff
@@ -239,3 +247,67 @@ def restart(self, request: web.Request) -> Awaitable[None]:
...
+                ATTR_CHANGELOG_PATH: f"/addons/{addon.slug}/changelog"
+                if addon.with_changelog
+                else None,
+                ATTR_UPDATE_PATH: f"/addons/{addon.slug}/update",
+                ATTR_PANEL_PATH: f"/update-available/{addon.slug}",
+                ATTR_VERSION: addon.version,
+                ATTR_VERSION_LATEST: addon.latest_version,
+            }
+            for addon in self.sys_addons.installed
+            if not addon.need_update
```
Comment (balloob) [inline supervisor/api/supervisor.py:310]: Should this be `if addon.need_update` ?

File: supervisor/api/supervisor.py [hunk 2]
```diff
@@ -239,3 +247,67 @@ def restart(self, request: web.Request) -> Awaitable[None]:
...
+            )
+
+        # OS
+        if self.sys_os.need_update:
+            available_updates.append(
+                {
+                    ATTR_UPDATE_TYPE: "os",
+                    ATTR_UPDATE_PATH: "/os/update",
+                    ATTR_PANEL_PATH: "/update-available/os",
+                    ATTR_CHANGELOG_URL: f"https://github.com/home-assistant/operating-system/compare/{self.sys_os.version}...{self.sys_os.latest_version}",
```
Comment (balloob) [inline supervisor/api/supervisor.py:289]: I think that we don't need to include the changelog URL or update path. Instead aim to use the API output to render a link to `panel_path` and have that page contain the changelog + update buttons etc.

File: supervisor/api/supervisor.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/api/supervisor.py:310]: Ah, yes all my addons where up to date, this was changed to test it, needs to be reverted indeed

File: supervisor/api/const.py [hunk 3]
```diff
@@ -10,3 +10,7 @@
 ATTR_USE_NTP = "use_ntp"
 ATTR_USE_RTC = "use_rtc"
 ATTR_APPARMOR_VERSION = "apparmor_version"
+ATTR_UPDATE_PATH = "update_path"
```
Comment (balloob) [inline supervisor/api/const.py:13]: The easiest way to keep it compatible and future proof is by having the supervisor host the "Update available" page and have that page do the update. In that case we don't need `update_path`.

**Review by balloob — APPROVED** [review]

**Pull request merged**